rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1278,6 +1279,26 @@ else if (activations.isActive(ConsensusRule.RSKIP146) && !activations.isActive(C
         }
     }
 
+    private void checkIfEntryExistsInTxsWaitingForSignatures(Keccak256 rskTxHash, BtcTransaction btcTransaction, Map<Keccak256, BtcTransaction> txsWaitingForSignatures) {
+        if (txsWaitingForSignatures.containsKey(rskTxHash)) {
+            logger.error(
+                "[checkIfEntryExistsInTxsWaitingForSignatures] An entry for the given rskTxHash {} already exists. Entry " +
+                    "overriding for txsWaitingForSignatures map is not allowed." +
+                "Existing btcTx value: {}" +
+                "btcTx value trying to be put: {}" +
+                rskTxHash,
+                txsWaitingForSignatures.get(rskTxHash),
+                btcTransaction
+            );
```

Comment:
I wouldn't add the BTC transactions here. Not sure it adds any value, just knowing that there's already an entry should be enough